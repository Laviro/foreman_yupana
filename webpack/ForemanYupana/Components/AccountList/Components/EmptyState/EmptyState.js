import React from 'react';
import { EmptyState } from 'patternfly-react';

const yupanaEmptyState = () => (
  <EmptyState>
    <EmptyState.Icon />
    <EmptyState.Title>Fetching data about your accounts</EmptyState.Title>
    <EmptyState.Info>Loading...</EmptyState.Info>
  </EmptyState>
);

export default yupanaEmptyState;
