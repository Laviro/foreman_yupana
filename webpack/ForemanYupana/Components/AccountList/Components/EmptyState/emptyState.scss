.blank-slate-pf {
  .spinner {
    float: none;
  }
}
