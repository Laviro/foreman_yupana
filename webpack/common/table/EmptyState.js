import React from 'react';
import PropTypes from 'prop-types';
import {
  EmptyState,
  EmptyStateIcon,
  Spinner,
  EmptyStateVariant,
  Title,
} from '@patternfly/react-core';
import { ExclamationCircleIcon, CheckIcon } from '@patternfly/react-icons';

import { STATUS } from 'foremanReact/constants';
import { translate as __, sprintf } from 'foremanReact/common/I18n';

const TableEmptyState = ({ status, error, rowsLength }) => {
  switch (status) {
    case STATUS.PENDING:
      return (
        <EmptyState variant={EmptyStateVariant.small}>
          <EmptyStateIcon variant="container" component={Spinner} />
          <Title headingLevel="h2" size="lg">
            {__('Loading')}
          </Title>
        </EmptyState>
      );
    case STATUS.ERROR:
      return (
        <EmptyState variant={EmptyStateVariant.small}>
          <EmptyStateIcon
            variant="container"
            component={ExclamationCircleIcon}
          />
          <Title headingLevel="h2" size="lg">
            {sprintf(__('The server returned the following error: %s'), error)}
          </Title>
        </EmptyState>
      );
    case STATUS.RESOLVED:
      if (rowsLength > 0) return null;
      return (
        <EmptyState variant={EmptyStateVariant.small}>
          <EmptyStateIcon
            variant="container"
            component={CheckIcon}
          />
          <Title headingLevel="h2" size="lg">
            {__('There are no remmediations for your hosts')}
          </Title>
        </EmptyState>
      );
    default:
      return null;
  }
};

TableEmptyState.propTypes = {
  status: PropTypes.string,
  error: PropTypes.string,
  rowsLength: PropTypes.number,
};

TableEmptyState.defaultProps = {
  status: '',
  error: '',
  rowsLength: 0,
};

export default TableEmptyState;
