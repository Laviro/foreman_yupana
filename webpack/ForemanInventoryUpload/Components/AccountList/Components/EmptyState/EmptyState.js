import React from 'react';
import { EmptyState, Spinner } from 'patternfly-react';
import './emptyState.scss';

const inventoryEmptyState = () => (
  <EmptyState>
    <Spinner loading inline size="lg" />
    <EmptyState.Title>Fetching data about your accounts</EmptyState.Title>
    <EmptyState.Info>Loading...</EmptyState.Info>
  </EmptyState>
);

export default inventoryEmptyState;
