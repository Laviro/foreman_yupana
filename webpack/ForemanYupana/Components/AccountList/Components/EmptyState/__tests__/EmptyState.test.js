import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import EmptyState from '../EmptyState';

const fixtures = {
  'render without Props': {},
  /** fixtures, props for the component */
};

describe('EmptyState', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(EmptyState, fixtures));
});
