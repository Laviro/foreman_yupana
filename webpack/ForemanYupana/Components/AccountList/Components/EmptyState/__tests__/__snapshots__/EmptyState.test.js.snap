// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState rendering render without Props 1`] = `
<EmptyState
  className=""
>
  <EmptyStateIcon
    className=""
    name="add-circle-o"
    type="pf"
  />
  <EmptyStateTitle
    className=""
  >
    Fetching data about your accounts
  </EmptyStateTitle>
  <EmptyStateInfo
    className=""
  >
    Loading...
  </EmptyStateInfo>
</EmptyState>
`;
